dask/dataframe/shuffle.py

```diff
@@ -239,6 +239,13 @@ def set_index(
             M.set_index, index, align_dataframes=False, drop=drop, **kwargs
         ).clear_divisions()
 
+    if compute is not False:
```

Comment:
Can we change compute to no default to see if this is given at all? We would also start raising if compute=False in the future

Reply:
Done https://github.com/dask/dask/pull/10784/commits/4d288ea1dca12627a81a68029bb05542ad7992d9